internal/rpmmd/repository.go

```diff
@@ -150,6 +156,39 @@ func LoadRepositories(confPaths []string, distro string) (map[string][]RepoConfi
 		return nil, err
 	}
 
+	if strings.HasPrefix(distro, "rhel-8") {
```

Comment:
Please don't hard code `rhel-8` here. This module has otherwise no knowledge of the names of distributions. If we ever add another one that needs subscriptions enabled (`rhel-9`, maybe? :wink:), it will be hard to find all the places that we've special-cased names.

Instead, pass this information through `RepoConfig`. I think there's no way around splitting it into two structs—one for reading from `/usr/share/osbuild-composer/repositories`, that contains `"rhsm": true`, and one for passing to `dnf-json`, that contains the `ssl*` fields.

Reply:
I have added the rhsm field to the repo files and use that to determine if a repo needs rhsm certs.